sensum-ethereum/test/sensations-test.js

```diff
@@ -1,20 +1,32 @@
 const { expect } = require("chai");
 const { ethers } = require("hardhat");
+const legacySensations = require("./legacy-sensations.json");
 
-// FIXME: Use Sensations contract!
-describe("Greeter", function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const Greeter = await ethers.getContractFactory("Greeter");
-    const greeter = await Greeter.deploy("Hello, world!");
-    await greeter.deployed();
+const checkSensation = async (contract, sensation, expectedLength) => {
+  const tx = await contract.newSensation(sensation);
+  const receipt = await tx.wait();
+  const eventArgs = receipt.events.find(({ event }) => event === "Synapsis")
+    .args[0];
 
-    expect(await greeter.greet()).to.equal("Hello, world!");
+  expect(await contract.getSensationsLength()).to.equal(expectedLength);
+  expect(eventArgs[1]).to.equal(sensation.message);
+  expect(parseInt(eventArgs[0]._hex, 16)).to.equal(sensation.avatar);
+};
 
-    const setGreetingTx = await greeter.setGreeting("Hola, mundo!");
-
-    // wait until the transaction is mined
-    await setGreetingTx.wait();
-
-    expect(await greeter.greet()).to.equal("Hola, mundo!");
+describe("Sensum", function () {
+  let Sensations;
+  let sensations;
+  beforeEach(async function () {
+    Sensations = await ethers.getContractFactory("Sensations");
+    sensations = await Sensations.deploy();
+  });
+  it("Should have 0 sensations in the begining", async () =>
+    expect(await sensations.getSensationsLength()).to.equal(0));
+  it("Should be able to send a sensation", () =>
+    checkSensation(sensations, { avatar: 0, message: "M o n i" }, 1));
+  it("Should be able to handle all the legacy sensations", async function () {
```

Comment:
nice! So one single contract was able to handle all the legacy sensations, I mean in terms of "available" space, right? :rocket:

Reply:
Yes, looks like it